### PR TITLE
feat(docs): add coc-rust-analyzer

### PR DESCRIPTION
**Pull request by fannheyward**

### CODE REVIEW

**Comment by matklad** [comment]
r? @mark-i-m 

bors delegate=mark-i-m

**Comment by matklad** [comment]
bors r=JeanMertz

File: docs/user/README.md [hunk 1]
```diff
@@ -104,35 +104,12 @@ to load path and require it in `init.el`
...
- "languageserver": {
-  "rust": {
-    "command": "ra_lsp_server",
-    "filetypes": ["rust"],
-    "rootPatterns": ["Cargo.toml"]
-  }
- }
-}
-```
+* Run `:CocInstall coc-rust-analyzer` to install `coc-rust-analyzer`, this extension already implemented _almost_ all features supported in VSCode extension:
```
Comment (JeanMertz) [inline docs/user/README.md]:
```suggestion
* Run `:CocInstall coc-rust-analyzer` to install [coc-rust-analyzer], this extension implements _most_ of the features supported in the VSCode extension:
```

File: docs/user/README.md [hunk 2]
```diff
@@ -104,35 +104,12 @@ to load path and require it in `init.el`
...
+  - same configurations as VSCode extension, `rust-analyzer.raLspServerPath`, `rust-analyzer.enableCargoWatchOnStartup` etc.
+  - same commands too, `rust-analyzer.analyzerStatus`, `rust-analyzer.startCargoWatch` etc.
+  - highlighting and inlay_hints is not works by now
 
 [coc.nvim]: https://github.com/neoclide/coc.nvim
-[coc-wiki]: https://github.com/neoclide/coc.nvim/wiki/Language-servers#rust
 [coc-vim-conf]: https://github.com/neoclide/coc.nvim/#example-vim-configuration
```
Comment (JeanMertz) [inline docs/user/README.md]:
```suggestion
[coc-rust-analyzer]: https://github.com/fannheyward/coc-rust-analyzer
```

File: docs/user/README.md [hunk 3]
```diff
@@ -104,35 +104,12 @@ to load path and require it in `init.el`
...
-    "filetypes": ["rust"],
-    "rootPatterns": ["Cargo.toml"]
-  }
- }
-}
-```
+* Run `:CocInstall coc-rust-analyzer` to install `coc-rust-analyzer`, this extension already implemented _almost_ all features supported in VSCode extension:
+  - same configurations as VSCode extension, `rust-analyzer.raLspServerPath`, `rust-analyzer.enableCargoWatchOnStartup` etc.
+  - same commands too, `rust-analyzer.analyzerStatus`, `rust-analyzer.startCargoWatch` etc.
+  - highlighting and inlay_hints is not works by now
```
Comment (JeanMertz) [inline docs/user/README.md]:
```suggestion
  - highlighting and inlay_hints is not yet implemented
```

**Comment by mark-i-m** [comment]
Ah, sorry, I missed the first ping. I haven't tried this plugin and don't know the author (or their trustworthiness), but it looks nifty.

**Comment by JeanMertz** [comment]
Just as an FYI: I don't know the author either, but did vet the code, and they are very fast in adding new features to get it up to par with the vscode plugin. I've get to run into any major issues.

I thought of rolling my own plugin, but then this one came along, and it's been working great so far. Definitely recommended for Vim users looking into using RA.

**Pull request merged**